contracts/InterchainTokenFactory.sol

```diff
@@ -153,19 +172,84 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
         }
     }
 
+    /**
+     * @notice Allow the minter to approve the deployer for a remote interchain token deployment that uses a custom destinationMinter address.
+     * This ensures that a token deployer can't choose the destinationMinter itself, and requires the approval of the minter to reduce trust assumptions on the deployer.
+     */
+    function approveDeployRemoteInterchainToken(
+        address deployer,
+        bytes32 salt,
+        string calldata destinationChain,
+        bytes calldata destinationMinter
+    ) external {
+        bytes32 tokenId = interchainTokenId(deployer, salt);
+        IInterchainToken token = IInterchainToken(interchainTokenService.interchainTokenAddress(tokenId));
+        if (!token.isMinter(msg.sender)) revert InvalidMinter(msg.sender);
+
+        bytes32 approvalKey = _deployApprovalKey(
+            DeployApproval({
+                minter: msg.sender,
+                deployer: deployer,
+                salt: salt,
+                destinationChain: destinationChain,
+                destinationMinter: destinationMinter
+            })
+        );
+
+        _interchainTokenFactoryStorage().deployApprovals[approvalKey] = true;
+
+        emit ApprovedDeployRemoteInterchainToken(msg.sender, deployer, salt, destinationChain, destinationMinter);
+    }
+
+    function _deployApprovalKey(DeployApproval memory approval) internal pure returns (bytes32 key) {
+        key = keccak256(abi.encode(PREFIX_DEPLOY_APPROVAL, approval));
+    }
+
+    function _useDeployApproval(DeployApproval memory approval) internal {
+        bytes32 approvalKey = _deployApprovalKey(approval);
+
+        InterchainTokenFactoryStorage storage slot = _interchainTokenFactoryStorage();
+
+        if (!slot.deployApprovals[approvalKey]) revert RemoteDeploymentNotApproved();
+
+        slot.deployApprovals[approvalKey] = false;
+    }
+
+    /**
+     * @notice Deploys a remote interchain token on a specified destination chain.
+     * @param salt The unique salt for deploying the token.
+     * @param minter The address to use as the minter of the deployed token on the destination chain. If the destination chain is not EVM,
+     * then use the more generic `deployRemoteInterchainToken` function below that allows setting an arbitrary destination minter that was approved by the current minter.
+     * @param destinationChain The name of the destination chain.
+     * @param gasValue The amount of gas to send for the deployment.
+     * @return tokenId The tokenId corresponding to the deployed InterchainToken.
+     */
+    function deployRemoteInterchainToken(
```

Comment:
Avoid adding the same function name with different params. I know the compiler allows you to, but I find it very confusing for readability.